sqlmesh/core/context.py

```diff
@@ -601,8 +641,13 @@ def run(
 
         if not self._loaded:
             # Signals should be loaded to run correctly.
-            with sys_path(*self.configs):
-                self._loader.load_signals(self)
+            if self._sqlmesh_loader:
+                with sys_path(*self.sqlmesh_configs):
```

Comment:
There's a bunch of these `with sys_path...` conditions that appear to do the same thing (call a known method on the loader that is the same between loaders).

I wonder if there's a way to make this more concise to reduce the duplication

Reply:
right i'll try to think of how to refactor these without making it too unclear